dataset/base_data_module.py

```diff
@@ -72,12 +91,17 @@ def test_dataloader(self):
             drop_last=True
         )
 
-    @staticmethod
-    def _collate(batch):
+    def _collate(self, batch):
         # batch contains a list of tuples of structure (sequence, target)
         a = pad_sequence([item["a_encoding"].squeeze() for item in batch], batch_first=True)
         b = pad_sequence([item["b_encoding"].squeeze() for item in batch], batch_first=True)
```

Comment:
Since you will iterate through positions in sequence it's better to use tensors of shape `[seq len; batch size; h]`. Or `1` instead of `h` it's happened before embedding. This may speed up computations, especially on GPU.